src/cmd/troubleshoot/image_helper.go

```diff
@@ -0,0 +1,125 @@
+package troubleshoot
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+var (
+	removeSchemaRegex      = regexp.MustCompile("^.*//(.*)$")
+	removeApiEndpointRegex = regexp.MustCompile("^(.*)/[^/]*$")
+	registryRegex          = regexp.MustCompile(`^(.*)/linux.*$`)
+	imageRegex             = regexp.MustCompile(`^.*/(linux.*)$`)
+)
+
+type imageInfo struct {
+	registry string
+	image    string
+	version  string
+}
+
+func splitImageName(imageName string) (imageInfo, error) {
+	// some image path examples that work with this function
+	//   aaa00000.dynatrace.com/linux/oneagent
+	//   aaa00000.dynatrace.com/linux/activegate:1.123
+
+	imgInfo := imageInfo{}
+
+	registryMatches := registryRegex.FindStringSubmatch(imageName)
+	if len(registryMatches) < 2 {
+		return imageInfo{}, fmt.Errorf("invalid image - registry not found (%s)", imageName)
+	}
+	imgInfo.registry = registryRegex.FindStringSubmatch(imageName)[1]
+
+	imageMatches := imageRegex.FindStringSubmatch(imageName)
+	if len(imageMatches) < 2 {
+		return imageInfo{}, fmt.Errorf("invalid image - endpoint not found (%s)", imageName)
+	}
+	imgInfo.image = imageRegex.FindStringSubmatch(imageName)[1]
```

Comment:
```suggestion
	imgInfo.image = imageMatches[1]
```
If you already use submatches, the registryRegex and imageRegex could be combined so FindStringSubmatch must only be called once.
I.e.

```
regex = regexp.MustCompile(`^(.*)/(linux.*)$`)

parsedUri := regex.FindStringSubmatch(imageName)
imgInfo.registry = parsedUri[1]
imgInfo.image = parsedUri[2]
```